spec/helpers/application_helper_spec.rb

```diff
@@ -12,4 +12,26 @@
       expect(html).to have_xpath("//span[@aria-label='#{tooltip_text}']")
     end
   end
+
+  describe '#decorated_session' do
+    context 'with service provider' do
+      it 'returns the service provider decorator' do
+        @sp_name = 'any sp'
+        decorator = double
+        allow(ServiceProviderSessionDecorator).to receive(:new).and_return(decorator)
+
+        expect(helper.decorated_session).to eq decorator
+      end
+    end
+
+    context 'without service provider' do
+      it 'returns the regular sessiin decorator' do
+        @sp_name = nil
+        decorator = double
+        allow(SessionDecorator).to receive(:new).and_return(decorator)
```

Comment:
sure, that's fine with me. I usually don't do type checking because it feels so non-OO to me...but in this case it is essentially the same thing AND shorter, so why not?